src/tests/class_tests/smartpeak/source/ParametersParser_test.cpp

```diff
@@ -17,92 +17,88 @@
-// $Maintainer: Douglas McCloskey $
+// $Maintainer: Douglas McCloskey, Ahmed Khalil $
 // $Authors: Douglas McCloskey $
 // --------------------------------------------------------------------------
 
+#include <gtest/gtest.h>
 #include <SmartPeak/test_config.h>
-
-#define BOOST_TEST_MODULE ParametersParser test suite
-#include <boost/test/included/unit_test.hpp>
 #include <SmartPeak/io/ParametersParser.h>
 
 using namespace SmartPeak;
 using namespace std;
 
-BOOST_AUTO_TEST_SUITE(parameterparser)
-
-BOOST_AUTO_TEST_CASE(read)
+TEST(ParametersParser, read)
 {
   const string pathname = SMARTPEAK_GET_TEST_DATA_PATH("FileReader_parameters.csv");
   ParameterSet parameters;
   ParametersParser::read(pathname, parameters);
-  BOOST_CHECK_EQUAL(parameters.size(), 3);
+  EXPECT_EQ(parameters.size(), 3);
 
   const string func1 {"func1"};
 
-  BOOST_REQUIRE(parameters.count(func1));
+  EXPECT_TRUE(parameters.count(func1));
   auto function_parameters = parameters.at(func1);
-  BOOST_REQUIRE(function_parameters.size() == 3);
-  BOOST_CHECK_EQUAL(function_parameters[0].getName(), "param1");
-  BOOST_CHECK_EQUAL(function_parameters[0].getType(), "int");
-  BOOST_CHECK_EQUAL(function_parameters[0].getValueAsString(), "-1");
-  BOOST_CHECK_EQUAL(function_parameters[0].getDescription(), "");
-
-  BOOST_CHECK_EQUAL(function_parameters[1].getName(), "param2");
-  BOOST_CHECK_EQUAL(function_parameters[1].getType(), "float");
-  BOOST_CHECK_EQUAL(function_parameters[1].getValueAsString(), "-1");
-  BOOST_CHECK_EQUAL(function_parameters[1].getDescription(), "a description");
-
-  BOOST_CHECK_EQUAL(function_parameters[2].getName(), "param3");
-  BOOST_CHECK_EQUAL(function_parameters[2].getType(), "float");
-  BOOST_CHECK_EQUAL(function_parameters[2].getValueAsString(), "-1");
-  BOOST_CHECK_EQUAL(function_parameters[2].getDescription(), "a quoted description, containing a comma");
+  EXPECT_TRUE(function_parameters.size() == 3);
+  EXPECT_STREQ(function_parameters[0].getName().c_str(), "param1");
+  EXPECT_STREQ(function_parameters[0].getType().c_str(), "int");
+  EXPECT_STREQ(function_parameters[0].getValueAsString().c_str(), "-1");
+  EXPECT_STREQ(function_parameters[0].getDescription().c_str(), "");
+
+  EXPECT_STREQ(function_parameters[1].getName().c_str(), "param2");
+  EXPECT_STREQ(function_parameters[1].getType().c_str(), "float");
+  EXPECT_STREQ(function_parameters[1].getValueAsString().c_str(), "-1");
+  EXPECT_STREQ(function_parameters[1].getDescription().c_str(), "a description");
+
+  EXPECT_STREQ(function_parameters[2].getName().c_str(), "param3");
+  EXPECT_STREQ(function_parameters[2].getType().c_str(), "float");
+  EXPECT_STREQ(function_parameters[2].getValueAsString().c_str(), "-1");
+  EXPECT_STREQ(function_parameters[2].getDescription().c_str(), "a quoted description, containing a comma");
 
   const string func2 {"func2"};
 
-  BOOST_REQUIRE(parameters.count(func2));
```

Comment:
This is quite minor, but BOOST_REQUIRE should be replaced by some ASSERT_* . The aim of using BOOST_REQUIRE is to stop the test as we will get into severe trouble and the rest of the test does not make sense (null pointer, out of range..) which ASSERT_ seems to be the counter part in google test. by the way, BOOST_REQUIRE was not so used in SmartPeak but I think it's a good practice since it add some readability to the test execution, which I think should not be lost, if possible.

Reply:
i'm aware of that